File: java/src/main/java/org/sparky/model/builders/FilterBuilder.java
package org.sparky.model.builders;

import org.sparky.model.Filter;
import org.sparky.model.Rule;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by lloyd on 19/10/2015.
 */
public class FilterBuilder {

    private VariableBuilder left;
    private VariableBuilder right;

    public FilterBuilder withLeft(VariableBuilder variableBuilder){
        left = variableBuilder;
        return this;
    }

    public FilterBuilder withRight(VariableBuilder variableBuilder){
        right = variableBuilder;
        return this;
    }

    public String columnName(){
        return left.buildAsColumnName();
    }

    public Rule getRule() {
        return right.build();
    }
}

File: net/Sparky/Model/Builder/FilterBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sparky.Model.Builder
{
    public class FilterBuilder
    {
        private VariableBuilder _left;
        private VariableBuilder _right;

        public FilterBuilder withLeft(VariableBuilder left)
        {
            _left = left;
            return this;
        }

        public FilterBuilder withRight(VariableBuilder right)
        {
            _right = right;
            return this;
        }

        public String ColumnName()
        {
            return _left.BuildAsColumnName();
        }

        public Rule GetRule()
        {
            return _right.Build();
        }
    }
}
